### PR TITLE
🎉 Create a Helm Chart For Airbyte

### PR DESCRIPTION
**Pull request by jonstacks**
## What

See number #1868. This creates an initial helm chart for installing Airbyte in Kubernetes to make it easier for users who are more familiar with helm. It also includes GitHub actions to help continually test that the chart works in the most basic case.
All of the templates are based off of the kustomize folder, but `minio` and `postgres` have been removed in favor of adding the bitnami helm charts as dependencies since they have an active community and allow easily tweaking their install.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

This does not fall under any of the existing checklist items regarding connectors.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
thanks for the contribution @jonstacks this is amazing!

**Comment by michel-tricot** [comment]
Thank you @jonstacks !!

**Comment by marcosmarxm** [comment]
@jonstacks this is an incredible contribution and complex one! thanks a lot again. Davin and Jared will review this week your code. There is anything you feel needs attention before they took a look or I can help you?

**Comment by jonstacks** [comment]
@marcosmarxm, the only thing I can think of is that the environment variables which get stored in the configmap may not be correct for every possible combination of how a user is deploying minio, postgres, etc. Those should be easy fixes going forward though when they are encountered. When something new does come up, we can add a matrix of test cases(github actions) to make sure there are no regressions going forward once we fix an issue.

**Comment by danieldiamond** [comment]
Trying to run this but running into PV issues - is this missing a `PersistentVolume` configuration?

**Comment by jonstacks** [comment]
@danieldiamond,

Someone in the slack chat also brought this up. Looks like its caused by defaulting the `server.persistence.stoargeClass` to `standard`, when it should be `""`(will use the default storage class). I'll get a fix out for it. In the meantime, you can supply:

```
server:
  persistence:
    storageClass: ""
```

to the current chart and it should use the default storage class. You might have to delete the current pvc if you get an error installing

File: charts/airbyte/Makefile
Comment (jrhizor) [inline charts/airbyte/Makefile:1]: nit: I'd prefer for this to be a bash script instead of a `Makefile`. It'd be nice for people to be able to interface with this without installing xcode tools on Mac, for example.

File: charts/airbyte/Makefile
Comment (jrhizor) [inline charts/airbyte/Makefile:2]:
```suggestion
NAMESPACE := airbyte
```

Maybe `airbyte` so this can be used for diagnostics/install for non-test cases more easily?

File: charts/airbyte/Makefile
Comment (jrhizor) [inline charts/airbyte/Makefile:14]:
```suggestion
				--timeout 600s \
```

We've seen some long download times for some users doing something locally.

File: charts/airbyte/templates/env-configmap.yaml
Comment (jrhizor) [inline charts/airbyte/templates/env-configmap.yaml:1]: We should split this into a config map and secrets map (like https://github.com/airbytehq/airbyte/pull/5962)

Comment (jrhizor) [reply charts/airbyte/templates/env-configmap.yaml:1]: Also, should we be making all of these configurable from the values file @davinchia ?

File: charts/airbyte/templates/gcs-log-creds-secret.yaml
Comment (jrhizor) [inline charts/airbyte/templates/gcs-log-creds-secret.yaml:1]: related to the comment on `charts/airbyte/templates/env-configmap.yaml`

File: charts/airbyte/templates/scheduler/deployment.yaml
Comment (jrhizor) [inline charts/airbyte/templates/scheduler/deployment.yaml:147]: These should be secret mounts as well.

File: charts/airbyte/templates/temporal/configmap.yaml
Comment (jrhizor) [inline charts/airbyte/templates/temporal/configmap.yaml:1]: I don't know how easy this is to do, but in theory we could use https://github.com/temporalio/helm-charts, configured to use our dynamicconfig  + postgres.

Not blocking, but something for us to consider.

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:33]: We'll also need to add this file to the bumpversion config so we catch these image tags.

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:104]: This isn't safe to have >1 at the moment. In the future we'll add support for scaling up schedulers. We could either remove the configurability or just add a comment saying this should always be 1.

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:156]: @davinchia we should pick a specific version for this instead of using `latest`. We do this in the Kustomize version as well.

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:199]: Same for the scheduler comment above. We're planning on making this horizontally scalable but haven't yet.

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:213]:
```suggestion
  ## Configure extra options for the server containers' liveness and readiness probes
```

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:276]: I think you mentioned that you're planning on changing this away from `standard`?

File: charts/airbyte/templates/env-configmap.yaml
Comment (danieldiamond) [inline charts/airbyte/templates/env-configmap.yaml:35]: Should this be `http://airbyte-minio-svc:9000`? 
Ran into issues where it couldn't execute to this endpoint

File: charts/airbyte/values.yaml
Comment (jrhizor) [inline charts/airbyte/values.yaml:277]: @davinchia we definitely want a larger storage here. Maybe that should be in a separate "production" values file?

File: charts/airbyte/templates/NOTES.txt
Comment (jrhizor) [inline charts/airbyte/templates/NOTES.txt:1]: Nice!

**Review by jrhizor — APPROVED** [review]
This is fantastic!

I left a bunch of comments. All of the ones to @davinchia are forward-looking and can easily be done after this PR is merged. 

The one thing that we really need to do before merging is add the couple files to bumpversion with the latest version already.

File: .github/workflows/helm.yaml
Comment (jonstacks) [reply .github/workflows/helm.yaml:69]: I ran this helm action in my fork and can try to move to `helm/kind-action`. This is just the one I was familiar with for bootstrapping a k8s cluster.

File: charts/airbyte/values.yaml
Comment (jonstacks) [reply charts/airbyte/values.yaml:276]: @jrhizor, yes. This was a mistake on my part. I'll have a patch out for it shortly. In our EKS cluster we have a storage class called `standard` and that is also the name of the storage class in the kind cluster I was using for local dev so  I didn't hit this issue sooner.

```
kubectl get storageclass
NAME                 PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
standard (default)   rancher.io/local-path   Delete          WaitForFirstConsumer   false                  3m56s
```

Comment (jonstacks) [reply charts/airbyte/values.yaml]: This should be fixed in https://github.com/airbytehq/airbyte/pull/5891/commits/2df0ec4e75a563e6954e51ef6002bb82c4881e1a

File: charts/airbyte/Makefile
Comment (jonstacks) [reply charts/airbyte/Makefile:1]: I can make that change. Do you have a preference on what the script is called? I assume you'd just want commands like the following?

```sh
./script-name install
./script-name test
./script-name update-docs
```

File: charts/airbyte/values.yaml
Comment (jonstacks) [reply charts/airbyte/values.yaml:281]: From what I've seen just browsing the bitnami helm community, most projects usually leave the default pretty small(1-10Gi range). I think the reasoning is the same as not supplying resource request and limits so that it installs on things like mini kube or kind clusters. I think it would be up to the end user to maintain their own values files and apply them either with `-f production.yaml` or `--set server.persistence.size` when they install the chart.

File: charts/airbyte/values.yaml
Comment (jonstacks) [reply charts/airbyte/values.yaml]: Fixed in https://github.com/airbytehq/airbyte/pull/5891/commits/ab8a270db45b638bc113dd1b2ef8478f761ba7af

File: .github/workflows/helm.yaml
Comment (davinchia) [reply .github/workflows/helm.yaml:47]: @jrhizor created https://github.com/airbytehq/airbyte/issues/6131 to remind us to address some of these comments after we merge this in.

File: charts/airbyte/templates/env-configmap.yaml
Comment (davinchia) [reply charts/airbyte/templates/env-configmap.yaml:1]: Yes we should make this configurable. 

I can handle splitting this out in a follow up PR.

**Review by davinchia — APPROVED** [review]
Nice work!

Some question to make sure I understand and two things from me:
- can you make sure to add a newline at the end of every file?
- can you make the variables in the configmap injectable from the values.yaml?

I'll merge this in once you do and make some changes so this better fits in with our build system. 

Thank you for the contribution :)

File: charts/airbyte/templates/NOTES.txt
Comment (jrhizor) [reply charts/airbyte/templates/NOTES.txt:1]: https://helm.sh/docs/chart_template_guide/notes_files/ 

So after spinning up people can get instructions to connect.

File: charts/airbyte/values.yaml
Comment (jonstacks) [reply charts/airbyte/values.yaml:53]: @davinchia, hand written. I used a lot of the bitnami helm-charts as inspiration for writing the docs(ex: [airflow](https://github.com/bitnami/charts/blob/master/bitnami/airflow/values.yaml#L210)). They might have something special for that, but I didn't see anything.

File: .github/workflows/helm.yaml
Comment (jonstacks) [reply .github/workflows/helm.yaml:74]: @davinchia, This job in the workflow:

1. Spins up a kind cluster
2. Does a helm install of the chart with the default `values.yaml` file and waits(with the `--wait 600s` for all the pods to be ready according to the readiness probe. If there is a misconfig in the env-configmap, the pod should crash loop and the install will fail. I've seen this locally when it can't reach postgres or mino it just crash loops.
3. It then runs a `helm test` which spins up a test pod to make sure the web frontend `index.html` is able to be fetched. (If there are other good tests, like hitting specific routes on the server service, those can be added as well for a more comprehensive test)

You can find the an example and output of the action here(https://github.com/jonstacks/airbyte/runs/3613634997)

File: charts/airbyte/Makefile
Comment (jonstacks) [reply charts/airbyte/Makefile]: @jrhizor, I moved this to a `ci.sh` bash script and removed the Makefile.

File: charts/airbyte/templates/NOTES.txt
Comment (davinchia) [reply charts/airbyte/templates/NOTES.txt:1]: neat!

**Comment by davinchia** [comment]
Spoke with Jon - he is going to follow up on the configmap changes in a second PR to keep things manageable.

File: charts/airbyte/values.yaml
Comment (davinchia) [reply charts/airbyte/values.yaml:53]: amazing - thought they were generated!

**Pull request merged**
